template/src/ontology/Makefile.jinja2

```diff
@@ -1203,14 +1203,17 @@ update_docs:
 	mkdocs gh-deploy --config-file ../../mkdocs.yaml
 {%- endif %}
 
-# Note to future generations: prepending ./ is a safety measure to ensure that 
-# the environment does not malicously set `CLEANFILES` to `\`.
+# Note to future generations: computing the real path relative to the
+# current directory is a way to ensure we only clean up directories that
+# are located below the current directory, regardless of the contents of
+# the *DIR variables.
 .PHONY: clean
 clean:{% if project.use_dosdps %}
 	$(MAKE) pattern_clean{%- endif %}
-	[ -n "$(MIRRORDIR)" ] && [ $(MIRRORDIR) != "." ] && [ $(MIRRORDIR) != "/" ] && [ $(MIRRORDIR) != ".." ] && [ -d ./$(MIRRORDIR) ] && rm -rf ./$(MIRRORDIR)/*
-	[ -n "$(TMPDIR)" ] && [ $(TMPDIR) != "." ] && [ $(TMPDIR) != "/" ] && [ $(TMPDIR) != ".." ] && [ -d ./$(TMPDIR) ] && rm -rf ./$(TMPDIR)/*
-	[ -n "$(UPDATEREPODIR)" ] && [ $(UPDATEREPODIR) != "." ] && [ $(UPDATEREPODIR) != "/" ] && [ $(UPDATEREPODIR) != ".." ] && [ -d ./$(UPDATEREPODIR) ] && rm -rf ./$(UPDATEREPODIR)/*
+	for dir in $(MIRRORDIR) $(TMPDIR) $(UPDATEREPODIR) ; do      \
+		reldir=$$(realpath --relative-to=$$(pwd) $$dir) ;    \
```

Comment:
Just because I have never seen this before. So lets say someone maliciously sets `MIRRORDIR` to `/` or `~`. Lets further assume this is run outside the docker container (just for the argument, not because I think this is likely; someone might do it mistakenly). Is there no risk with `--relative-to` to delete the root directory?

Reply:
I just read your issue https://github.com/INCATools/ontology-development-kit/issues/1002

still not understand exactly how the --relative-to works technically..

Reply:
No. Let’s say `MIRRORDIR` is set to `/`. Then the call to `realpath` will set `reldir` to something like `../../../../` (the number of `../` dependent on how deep the current directory is compared to the root, e.g. if we are in `/home/users/alice/myontology/src/ontology/`, `reldir` will be `../../../../../../`, because the root is 6 directories above the current directory).

Then in the following line, we will fall into the first pattern of the `case..esac` construct: `case $reldir in .*|"") ;;` (because `reldir` will start with a dot), so the `rm` command will not get executed.

This generalizes to any scenario where `MIRRORDIR` is set to any directory that is _above_ the current directory (or that _is_ the current directory).

Reply:
If you’re concerned about the call to `realpath`, a slightly simpler approach would be something like that:

```sh
for dir in $(MIRRORDIR) $(TMPDIR) $(UPDATEREPODIR) ; do
    case $dir in .*|/*|"") ;; *) rm -rf $dir/* ;; esac
done
```

where the first pattern of the `case..esac` construct will catch

* any case where `$dir` points to any directory above the current directory (because it starts with `.`);
* any case where `$dir` is an absolute pathname (because it starts with `/` ; this includes the case where `$dir` _is_ `/`)
* the case where `$dir` is empty (so that we do not run a `rm -rf /*` command).

But contrary to the version with `realpath`, this will _not_ catch cases where a malicious user hides the fact that he is trying to go upward in the filesystem hierarchy by:

* first going downward, for example by setting `$dir` to `tmp/../../../../../`;
* creating a symlink that points to a upward directory, and setting `$dir` to it.

Using `realpath` would catch those, because `realpath` (1) canonicalizes the pathname to remove any superfluous `../`), and (2) resolves any symbolic links along the way.

Reply:
I am not concerned. Just wondering about any risks as I can personally not reed this line of code - technically I mean. I trust your judgement!